reframe/core/launchers/__init__.py

```diff
@@ -36,6 +36,14 @@ class JobLauncher(metaclass=_JobLauncherMeta):
     #: :default: ``[]``
     options = variable(typ.List[str], value=[])
 
+    #: Dictionary of environment variables to be passed via the job launcher
+    #: invocation. The keys are the variable names and the values are their
+    #: corresponding values.
```

Comment:
We should expand this mentioning which backends respect and properly interpret this variable.